.github/workflows/release.yaml

```
@@ -1,127 +1,32 @@
+      - name: "🔒 Generate token"
+        id: generate_token
+        uses: tibdex/github-app-token@v1
         with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-      
-      - name: Cross build
-        run: |
-          #!/usr/bin/env bash
-
-          docker buildx bake \
-            -f ./docker-bake.hcl \
-            --set build.args.GO_LDFLAGS="$GO_LDFLAGS" \
-            --set cross.tags=ghcr.io/${{ github.repository }}:${{ github.ref_name }} \
-            --push \
-            cross
-
-      - name: Install crane
-        if: startsWith(github.ref, 'refs/tags/')
-        uses: imjasonh/setup-crane@00c9e93efa4e1138c9a7a5c594acd6c75a2fbf0c # v0.3
-      - name: Output image digest
-        if: startsWith(github.ref, 'refs/tags/')
-        id: image_digest
-        run: echo "IMAGE_DIGEST=$(crane digest ${IMAGE_URI_TAG})" >> $GITHUB_OUTPUT
+          app_id: ${{ secrets.OS_GITHUB_APP_ID }}
+          private_key: ${{ secrets.OS_GITHUB_APP_PRIVATE_KEY }}
 
-  sign:
-    name: Sign image and generate sbom
-    runs-on: ubuntu-latest
-    needs: [build]
-    if: startsWith(github.ref, 'refs/tags/')
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # tag=v3
-      - name: Login to GitHub Container Registry
-        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a # v2.1.0
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Run Trivy in fs mode to generate SBOM
-        uses: aquasecurity/trivy-action@e5f43133f6e8736992c9f3c1b3296e24b37e17f2 # master
+      - name: "☁️  Checkout repository"
+        uses: actions/checkout@v3
         with:
-          scan-type: 'fs'
-          format: 'spdx-json'
-          output: 'spdx.sbom.json'
-      - name: Install cosign
-        uses: sigstore/cosign-installer@dd6b2e2b610a11fd73dd187a43d57cc1394e35f9 # main
-      - name: Sign image and sbom
-        run: |
-          #!/usr/bin/env bash
-          set -euo pipefail
-          cosign sign -a git_sha=$GITHUB_SHA ${IMAGE_URI_DIGEST} --yes
-          cosign attach sbom --sbom spdx.sbom.json ${IMAGE_URI_DIGEST}
-          cosign sign -a git_sha=$GITHUB_SHA --attachment sbom ${IMAGE_URI_DIGEST} --yes
-        shell: bash
-        env:
-          IMAGE_URI_DIGEST: ${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }}
-
-  provenance:
-    needs: [build]
-    if: startsWith(github.ref, 'refs/tags/')
-    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.6.0
-    with:
-      image: ${{ needs.build.outputs.image }}
-      digest: ${{ needs.build.outputs.digest }}
-      registry-username: ${{ github.actor }}
-    secrets:
-      registry-password: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0
```

This needs to use the token also with the GitHub app

```suggestion
          fetch-depth: 0
          GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
```

---

For `actions/checkout`? I thought that it by default would use the default github token which has permissions to checkout the code? https://github.com/actions/checkout#usage Does it need a generated token to create the release and everything?

---

For protected branches, yes. Similar to https://github.com/open-sauced/pizza-cli/issues/35#issuecomment-1691901233.

I found out about this when looking into https://github.com/cycjimmy/semantic-release-action#basic-usage.

---

Correct. The GitHub App authentication is used to checkout the code 

https://github.com/open-sauced/api/blob/beta/.github/workflows/release.yml#L155-L159

And to do the versioning to bypass PR requirements for the commit

https://github.com/open-sauced/api/blob/beta/.github/workflows/release.yml#L155-L159

---

Probably the critical bit we're not syncing up on is this:

https://github.com/open-sauced/release/blob/21cbedd1feeff1883a9ee824c596172c9bdd986a/release.config.js#L169-L187

which was changed to _not_ make a commit via `@symantic-release/git` if `SKIP_NPM_PUBLISH` is set. I.e., don't attempt to commit abunch of javascript artifacts if there's no NPM package that is being operated on. It all sort of ends up being abit different with this Go program: since there are zero long lasting build artifacts that need to be added to source control, there's no additional commit ontop of the default branch that needs to happen.

---

The elevated permissions for pushing the tags.

---

Ahh, I see 👍

---

Okay, the error originates from the git plugin which isn't being added here. But `git push --tags` command is the failing command. Which will have to run regardless. So not so sure. Maybe we'll find out when this release runs.

---

Hmmm I'm not sure either: but I'm _fairly_ confident it will work with the `content: write` permission 😅 I guess we'll just have to wait and see!!!!! 🚀